Amazon.QLDB.Driver.Tests/RetryHandlerTests.cs

```diff
@@ -118,10 +118,37 @@ public void RetriableExecute_RetryWithRecoverRegardlessLimit_Succeed()
             retry.Verify(r => r.Invoke(It.IsAny<int>()), Times.Exactly(6));
         }
 
+        [TestMethod]
+        public void RetriableExecute_RetryWithRecoverExceedItsLimit_Succeed()
+        {
+            var handler = (RetryHandler)QldbDriverBuilder.CreateDefaultRetryHandler(NullLogger.Instance, 7);
+
+            var func = new Mock<Func<int>>();
+            var recover = new Mock<Action>();
+            var retry = new Mock<Action<int>>();
+
+            var invalid = new InvalidSessionException("invalid session");
+
+            func.SetupSequence(f => f.Invoke())
+                .Throws(invalid)
+                .Throws(invalid)
+                .Throws(invalid)
+                .Throws(invalid)
+                .Throws(invalid)
+                .Throws(invalid)
+                .Throws(invalid)
+                .Throws(invalid);
+
+            Assert.AreEqual(invalid,
+                Assert.ThrowsException<InvalidSessionException>(
+                    () => handler.RetriableExecute(func.Object,
+                    Driver.RetryPolicy.Builder().WithMaxRetries(4).Build(), recover.Object, retry.Object)));
+        }
+
```

Comment:
Lets add some tests where

-  RetryWithRecover limit > RetryLimit in RetryPolicy 
-  RetryWithRecover limit < RetryLimit in RetryPolicy